packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/TypeFactory.cs

```diff
@@ -358,5 +361,16 @@ protected virtual IReadOnlyList<TypeProvider> CreateSerializationsCore(InputType
         {
             return [];
         }
+
+        private readonly struct EnumCacheKey
+        {
+            public InputEnumType EnumType { get; }
+            public TypeProvider? DeclaringType { get; }
+            public EnumCacheKey(InputEnumType enumType, TypeProvider? declaringType)
+            {
+                EnumType = enumType;
+                DeclaringType = declaringType;
+            }
+        }
```

Comment:
actually if we use this as key of dictionaries, `record struct` might be more efficient and simpler
But this is also fine to me